packages/flame/test/cache/assets_cache_test.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:flame/cache.dart';
+import 'package:flame/flame.dart';
+import 'package:flame_test/flame_test.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+
+import '../fixtures/fixture_reader.dart';
+
+class _AssetsCacheMock extends Mock implements AssetsCache {
+  int disposedCount = 0;
+
+  void dispose() {
+    disposedCount++;
+  }
+}
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  group('AssetsCache', () {
+    test('readFile', () async {
+      final assetsCache = AssetsCache(prefix: '');
+      final fileName = fixture('test_text_file.txt').path;
+      final file = await assetsCache.readFile(fileName);
+      expect(file, isA<String>());
+
+      expect(
+        () async =>
+            assetsCache.readBinaryFile(fixture('test_text_file.txt').path),
+        failsAssert('"$fileName" was previously loaded as a text file'),
+      );
+    });
+
+    test('readJson', () async {
+      final assetsCache = AssetsCache(prefix: '');
+      final file = await assetsCache.readJson(fixture('chopper.json').path);
+      expect(file, isA<Map<String, dynamic>>());
+    });
+
+    test('readBinaryFile', () async {
+      final assetsCache = AssetsCache(prefix: '');
+      final fileName = fixture('caveace.fa').path;
+      final file = await assetsCache.readBinaryFile(fileName);
+      expect(file, isA<Uint8List>());
+
+      expect(
+        () async => assetsCache.readFile(fixture('caveace.fa').path),
+        failsAssert('"$fileName" was previously loaded as a binary file'),
+      );
+    });
+
+    test('clear', () async {
+      final assetsCache = AssetsCache(prefix: '');
+
+      final fileName = fixture('test_text_file.txt').path;
+      final file = await assetsCache.readFile(fileName);
+      expect(file, isA<String>());
+
+      final assetsCacheMock = _AssetsCacheMock();
+      expect(assetsCacheMock.disposedCount, 0);
+      assetsCache.clear(fileName);
+      assetsCacheMock.dispose();
+
+      /// On cache clear for confirmation increment dispose count
+      expect(assetsCacheMock.disposedCount, 1);
+    });
+
+    test('clearCache', () async {
+      final assetsCache = AssetsCache(prefix: '');
+
+      final fileName = fixture('test_text_file.txt').path;
+      final file = await assetsCache.readFile(fileName);
+      expect(file, isA<String>());
+
+      final assetsCacheMock = _AssetsCacheMock();
+      expect(assetsCacheMock.disposedCount, 0);
+      assetsCache.clearCache();
+      assetsCacheMock.dispose();
+      expect(assetsCacheMock.disposedCount, 1);
+
```

Comment:
```suggestion
      // If all file was not clear from cache then it will not readBinaryFile
```
This should be a comment, not a dartdoc